feature/subscriber-attributes/src/main/java/com/revenuecat/purchases/subscriberattributes/SubscriberAttributesManager.kt

```diff
@@ -160,8 +173,44 @@ class SubscriberAttributesManager(
         applicationContext: Application,
         completion: (deviceIdentifiers: Map<String, String?>) -> Unit
     ) {
+        obtainingDeviceIdentifiersObservable.numberOfProcesses++
         deviceIdentifiersFetcher.getDeviceIdentifiers(applicationContext) { deviceIdentifiers ->
             completion(deviceIdentifiers)
+            obtainingDeviceIdentifiersObservable.numberOfProcesses--
+        }
+    }
+
+    // This was added to resolve: https://revenuecats.atlassian.net/browse/CSDK-433. To summarize it,
+    // setting advertising IDs happens on a non-main thread. If that process is delayed for any reason, the attributes
+    // won't be stored in the device cache in time for the sync with the backend. This is a mechanism added so we can
+    // wait until the device identifiers have been obtained before continuing with other processes.
```

Comment:
@aboedo @vegaro I still need to fix/add tests for this PR but wanted to ask you what you think of this approach first. Lmk if you have any concerns or suggestions over it!